File: 链表/合并两个链表.go
package 链表

//毫无难点，秒杀
func mergeInBetween(list1 *ListNode, a int, b int, list2 *ListNode) *ListNode {
	head1 := list1
	for i := 1; i < a; i++ {
		head1 = head1.Next
	}
	head2 := head1.Next
	head1.Next = list2

	for i := 0; i < b-a+1; i++ {
		head2 = head2.Next
	}
	for list2.Next != nil {
		list2 = list2.Next
	}
	list2.Next = head2
	return list1
}

File: 链表/合并两个链表.java
package 链表;

public class 合并两个链表 {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        var dummyHead = new ListNode();
        var tmp = dummyHead;
        int i = 0;
        while (i < a) {
            tmp.next = list1;
            tmp = tmp.next;
            list1 = list1.next;
            i++;
        }
        tmp.next = list2;
        while (tmp.next != null) tmp = tmp.next;
        while (i <= b) {
            list1 = list1.next;
            i++;
        }
        while (list1 != null) {
            tmp.next = list1;
            tmp = tmp.next;
            list1 = list1.next;
        }
        return dummyHead.next;
    }
}
